images.CI/build-image.ps1

```diff
@@ -0,0 +1,60 @@
+param(
+    [String] [Parameter (Mandatory=$true)] $Image,
+    [String] [Parameter (Mandatory=$true)] $ClientId,
+    [String] [Parameter (Mandatory=$true)] $ClientSecret,
+    [String] [Parameter (Mandatory=$true)] $GitHubFeedToken,
+    [String] [Parameter (Mandatory=$true)] $ResourcesNamePrefix,
+    [String] [Parameter (Mandatory=$true)] $Location,
+    [String] [Parameter (Mandatory=$true)] $ResourceGroup,
+    [String] [Parameter (Mandatory=$true)] $StorageAccount,
+    [String] [Parameter (Mandatory=$true)] $SubscriptionId,
+    [String] [Parameter (Mandatory=$true)] $TenantId,
+    [String] [Parameter (Mandatory=$true)] $VirtualNetworkName,
+    [String] [Parameter (Mandatory=$true)] $VirtualNetworkRG,
+    [String] [Parameter (Mandatory=$true)] $VirtualNetworkSubnet
+)
+
+$TemplatePath = (Get-ChildItem -Path "images" -Include "$Image.json" -Recurse -Depth 2).FullName
+if (-not $TemplatePath)
+{
+    Write-Error "'-Image' parameter is not valid. You have to specify correct image type."
+    exit 1
+}
+
+$TempResourceGroupName = "${ResourcesNamePrefix}_${Image}"
+$InstallPassword = [System.GUID]::NewGuid().ToString().ToUpper()
+
+packer validate -syntax-only $TemplatePath
+
+$SensitiveData = @(
+    'OSType',
+    'StorageAccountLocation',
+    'OSDiskUri',
+    'OSDiskUriReadOnlySas',
+    'TemplateUri',
+    'TemplateUriReadOnlySas',
+    ':  ->'
+)
+
+Write-Host "Build $Image VM"
+packer build    -var "capture_name_prefix=$ResourcesNamePrefix" `
```

Comment:
can you please double check that the variables expected as secrets are actually marked as such in the packer json files

Reply:
Actually, only `client_secret` and `github_token` are marked as `sensitive-variables` in packer templates.
Other variables, like `client_id`, `capture_name_prefix`, `resource_group`, `storage_account` and etc should not be marked as sensitive because it can  complicate life of our customers who will build image locally.

These variables won't be shown in our CI because they are marked as secret on AzDo level:
![image](https://user-images.githubusercontent.com/16715858/75326473-be9ed480-588b-11ea-9cbb-6910805e4468.png)

As for the autogenerate values like temporary resource group and machine id (values that we can't know in advance), we use filter to exclude them from logs (see 56-59 lines)